src/api/app/components/watchlist_component.rb

```diff
@@ -16,6 +16,7 @@ def initialize(user:, current_object:, bs_request: nil, package: nil, project: n
     @bs_requests = bs_requests
     @packages = packages
     @projects = projects
+    @watch_count = WatchedItem.where(watchable: @object_to_be_watched).count
```

Comment:
I think you need to pass the `@watch_count` to `WatchlistIconComponent` instead of `WatchlistComponent`, right?

Reply:
`WatchlistIconComponent` is inherited from `WatchlistComponent` and all the logic is inside parent class. `WatchlistIconComponent` class can be removed

Reply:
Indeed. `WatchlistIconComponent` inherits from `WatchlistComponent` and is the one in charge of displaying the button with button with the eye + text.

Reply:
@rubhanazeem don't compare the .rb files but the .html.haml ones

Reply:
> `WatchlistIconComponent` is inherited from `WatchlistComponent` and all the logic is inside parent class. `WatchlistIconComponent` class can be removed because it's actually empty

I didn't realize the inheritance...

Reply:
Dismiss my review, please